homeassistant/components/matter/adapter.py

```diff
@@ -126,43 +47,13 @@ def register_platform_handler(
         if len(self.platform_handlers) == len(DEVICE_PLATFORM):
             self._platforms_set_up.set()
 
-    async def load_data(self) -> dict | None:
-        """Load data."""
-        try:
-            return await self._store.async_load()
-        except ValueError:
-            # Exception happens when the stored data is not deserializable with new Matter models
-            data = await get_matter_fallback_store(
-                self.hass, self.config_entry
-            ).async_load()
-            # Remove the serialized matter data. Devices will be re-interviewed.
-            if data is not None:
-                del data["node_interview_version"]
-            return data
-
-    async def save_data(self, data: dict) -> None:
-        """Save data."""
-        await self._store.async_save(data)
-
-    def delay_save_data(self, get_data: Callable[[], dict]) -> None:
-        """Save data, but not right now."""
-        self._store.async_delay_save(get_data, 60)
-
-    def get_server_url(self) -> str:
-        """Get the server URL."""
-        return cast(str, self.config_entry.data[CONF_URL])
-
-    def get_client_session(self) -> aiohttp.ClientSession:
-        """Get the client session."""
-        return async_get_clientsession(self.hass)
-
-    def get_node_lock(self, nodeid: int) -> asyncio.Lock:
-        """Get the lock for a node."""
-        if nodeid not in self._node_lock:
-            self._node_lock[nodeid] = asyncio.Lock()
-        return self._node_lock[nodeid]
+    async def setup_nodes(self) -> None:
+        """Set up all existing nodes."""
+        await self._platforms_set_up.wait()
+        for node in await self.matter_client.get_nodes():
+            await self._setup_node(node)
 
-    async def setup_node(self, node: MatterNode) -> None:
+    async def _setup_node(self, node: MatterNode) -> None:
         """Set up an node."""
         await self._platforms_set_up.wait()
```

Comment:
We await this in `setup_nodes` too. Maybe remove it here?

Reply:
Yeah, and we can do the listening for new nodes only if all existing nodes are added.